Jenkinsfile

```diff
@@ -0,0 +1,131 @@
+def registry = '562955126301.dkr.ecr.eu-west-2.amazonaws.com'
+def regCredsId = 'ecr:eu-west-2:ecr-user'
+def kubeCredsId = 'awskubeconfig002'
+def imageName = 'ffc-demo-api-gateway'
+def testImageName = 'ffc-demo-api-gateway-test'
+def repoName = 'ffc-demo-api-gateway'
+def branch = ''
+def pr = ''
+def mergedPrNo = ''
+def containerTag = ''
+
+def getMergedPrNo() {
+    def mergedPrNo = sh(returnStdout: true, script: "git log --pretty=oneline --abbrev-commit -1 | sed -n 's/.*(#\\([0-9]\\+\\)).*/\\1/p'").trim()
+    return mergedPrNo ?: ''
+}
+
+def getVariables(repoName) {
+    // jenkins checks out a commit, rather than a branch
+    // use the git cli to get branch info for the commit
+    def branch = sh(returnStdout: true, script: 'git ls-remote --heads origin | grep $(git rev-parse HEAD) | cut -d / -f 3').trim()
+    // and the github API to get the current open PR for the branch.
+    // Note: This will cause issues if one branch has two open PRs
+    def pr = sh(returnStdout: true, script: "curl https://api.github.com/repos/DEFRA/$repoName/pulls?state=open | jq '.[] | select(.head.ref == \"$branch\") | .number'").trim()
+    def rawTag = pr == '' ? branch : "pr$pr"
+    def containerTag = rawTag.replaceAll(/[^a-zA-Z0-9]/, '-').toLowerCase()
+    return [branch, pr, containerTag,  getMergedPrNo()]
+}
+
+def buildProductionImage(name, suffix) {
+  sh 'docker image prune -f'
+  sh "docker-compose -p $name-$suffix build --force-rm --no-cache --pull $name"
+}
+
+def buildTestImage(name, suffix) {
+  sh 'docker image prune -f'
```

Comment:
I don't think this is necessary now we've added it as a nightly job.  Running this in a pipeline risks breaking any other pipelines that may be running.

Reply:
Thanks, I've removed that.